userlib.go

```diff
@@ -23,7 +23,7 @@ import (
 
 type UUID = uuid.UUID
 
-// RSA key size (in bytes)
+// RSA key size (in bits)
 const RSAKeySizeBytes = 2048
```

Comment:
the name of the variable itself is inconsistent with its current value. The name says bytes, but the value is in bits.

Options:
1. Change variable name to `RSAKeySizeBits`. Not backwards compatible, but students should be able to update with a simple find/replace.
2. Keep variable name and change value to `256`. Backwards compatible for students, but will require updating its usage on L219 and L274 to pass `RSAKeySizeBytes * 8` to the `rsa.GenerateKey()` calls.

I vote for option 1 to rename the variable because key sizes are typically discussed in bits, while block sizes are discussed in bytes. Also, this matches the Go documentation for [rsa.GenerateKey()](https://golang.org/pkg/crypto/rsa/#GenerateKey), which uses bits.

@peyrin, @nicholasngai, thoughts?

Reply:
Oops, I forgot to update the variable name. I agree with option 1 - I don't think students ever need to use the RSAKeySizeBytes variable, so it should be only a small handful of people who need to find-and-replace it.

Reply:
Though if we update the variable name, need to make sure any other references to it (possibly also in userlib_test.go?) are also updated

Reply:
Good call. I will update the variable name and make sure that all of the tests still pass.

Reply:
Just updated this

Reply:
Huh, I actually kind of prefer using bytes. It makes more sense if you ever actually want to reference the variable (I can’t think of any function that would take in a number of bits rather than bytes), and it harmonizes with using 16 bytes for AES key sizes, which is what you would pass into a random number generator, as well.

Reply:
Apparently [rsa.GenerateKey()](https://golang.org/pkg/crypto/rsa/#GenerateKey) uses bits

Reply:
Any objections to this? If not we'll merge it

Reply:
@peyrin  @nicholasngai 

Is there really any need to make this variable publicly available at all? Students can use `AESKeySizeBytes` when generating random bytes for AES keys, but they shouldn't have to ever use a constant when generating RSA keys, right? The userlib has the size of RSA keys hardcoded.

Is it easier to just rename the variable to `const rsaKeySizeBits` so that it makes sense semantically with [rsa.GenerateKey()](https://golang.org/pkg/crypto/rsa/#GenerateKey) and it is also not accessible outside of `userlib`?

Reply:
I've never seen anyone actually use this variable - it doesn't even correspond to the max size of RSA encryption, so there's no practical use for the number 2048 as far as I know. If changing it to const doesn't break anything, I'm cool with it